services/debugger/source/eventUploader.go

```diff
@@ -97,7 +94,7 @@ func RecordEvent(writeKey, eventBatch string) bool {
 	configSubscriberLock.RLock()
 	defer configSubscriberLock.RUnlock()
 	if !misc.Contains(uploadEnabledWriteKeys, writeKey) {
-		eventBatchData, _ := json.Marshal(eventBatch)
+		eventBatchData := []byte(eventBatch)
```

Comment:
what's going on here with bytes and strings? is there an easy explanation?

Reply:
It seemed redundant to do an unmarshal and marshal again for a byte array. I didn't think of any reason why could have this been done

Reply:
do we even need to convert string to bytes?

Reply:
`eventsCacheMap` accepts a byteArray. This is common across sources, destinations and transformations. At the remaining places, we are doing an marshal to store as a byte array